src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -109,14 +109,39 @@ protected Object readResolve() throws ObjectStreamException {
      * <p>It will be false for a node which still has active children, like a step with a running body.
      * It will also be false for something that has finished but is pending child node creation,
      * such as a completed fork branch which is waiting for the join node to be created.
-     * <p>This can only go from true to false and is a shortcut for {@link FlowExecution#isCurrentHead}.
+     * <p>This can only go from true to false.
+     * @deprecated Usually {@link #isActive} is what you want. If you really wanted the original behavior, use {@link FlowExecution#isCurrentHead}.
      */
-    @Exported
+    @Deprecated
     public final boolean isRunning() {
         return getExecution().isCurrentHead(this);
     }
 
     /**
+     * Checks whether a node is still part of the active part of the graph.
+     * Unlike {@link #isRunning}, this behaves intuitively for a {@link BlockStartNode}:
+     * it will be considered active until the {@link BlockEndNode} is added.
+     */
+    @Exported(name="running")
+    public final boolean isActive() {
+        if (this instanceof FlowEndNode) { // cf. JENKINS-26139
+            return false;
+        }
+        List<FlowNode> currentHeads = exec.getCurrentHeads();
+        if (currentHeads.contains(this)) {
+            return true;
+        }
+        if (this instanceof BlockStartNode) {
+            for (FlowNode headNode : currentHeads) {
+                if (new LinearBlockHoppingScanner().findFirstMatch(headNode, Predicates.equalTo(this)) != null) {
```

Comment:
Ref: https://github.com/jenkinsci/workflow-support-plugin/pull/19#discussion_r84333212 for the same class of problem cropping up again. 

If we don't provide a comprehensive lookup for block info, the next best thing we can do is provide something that scans along nodes within a branch before going beyond the ParallelStartNode.  Basically like a  `ForkScanner`  mostly but without so strictly enforcing iteration order.   Might be easier now that we have trivial ways to check if a node is a ParallelStartNode.